src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,940 @@
+   elseif G.descr==Symbol("SO+") G.X=GAP.Globals.SO(1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("Omega+") G.X=GAP.Globals.Omega(1,G.deg,Int(order(G.ring)))
+   elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1,G.deg,G.ring_iso.codomain)
+   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.Omega(-1,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GO G.X=GAP.Globals.GO(0,G.deg,G.ring_iso.codomain)
+   elseif G.descr==:SO G.X=GAP.Globals.SO(0,G.deg,G.ring_iso.codomain)
+   elseif G.descr==:Omega G.X=GAP.Globals.Omega(0,G.deg,Int(order(G.ring)))
+   elseif G.descr==:GU G.X=GAP.Globals.GU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   elseif G.descr==:SU G.X=GAP.Globals.SU(G.deg,Int(characteristic(G.ring)^(div(degree(G.ring),2) ) ))
+   end
+end
+
+function Base.getproperty(G::MatrixGroup, sym::Symbol)
+
+   if sym === :mat_iso || sym === :ring_iso
+      if isdefined(G,sym) return getfield(G,sym) end
+      fm = gen_mat_iso(G.deg, G.ring)
+      G.ring_iso=fm.fr
+      G.mat_iso=fm
+
+   elseif sym === :X
+      if isdefined(G,:X) return getfield(G,:X) end
+      if isdefined(G,:descr)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         assign_from_description(G)
+      elseif isdefined(G,:gens)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         V = GAP.julia_to_gap([g.X for g in G.gens])
+         G.X=GAP.Globals.Group(V)
+      end
+
+   elseif sym === :gens
+      if isdefined(G, :gens) return getfield(G, :gens) end
+      if !isdefined(G, :X)
+         if !isdefined(G,:ring_iso)
+            fm = gen_mat_iso(G.deg, G.ring)
+            G.ring_iso=fm.fr
+            G.mat_iso=fm
+         end
+         assign_from_description(G)
+      end
+      L = GAP.Globals.GeneratorsOfGroup(getfield(G,:X))
+      G.gens=[MatrixGroupElem(G,G.mat_iso(L[i]),L[i]) for i in 1:length(L)]
+
+   else
+      return getfield(G, sym)
+
+   end
+
+end
+
+
+# add field :X
+function Base.getproperty(x::MatrixGroupElem, sym::Symbol)
+   if sym === :X && !isdefined(x,:X)
+      if !isdefined(x.parent,:ring_iso)
+         fm = gen_mat_iso(x.parent.deg, x.parent.ring)
+         setfield!(x.parent,:ring_iso,fm.fr)
+         setfield!(x.parent,:mat_iso,fm)
+      end
+      setfield!(x, :X, x.parent.mat_iso(x.elm))
+   elseif sym == :elm && !isdefined(x, :elm)                        # assuming that x.X, hence x.parent.ring_iso, are defined
+      setfield!(x, :elm, x.parent.mat_iso(x.X))
+   end
+   return getfield(x,sym)
+end
+
+Base.IteratorSize(::Type{<:MatrixGroup}) = Base.SizeUnknown()
+
+function Base.iterate(G::MatrixGroup)
+  L=GAP.Globals.Iterator(G.X)
+  if GAP.Globals.IsDoneIterator(L)
+    return nothing
+  end
```

Comment:
Groups are never empty, so how could this ever happen?
```suggestion
  @assert ! GAP.Globals.IsDoneIterator(L)
```

Reply:
I asked myself this questions too, but I just copied the procedure I used for generic groups too. Is it the case to correct in the file `GAPGroups.jl` too?